diff --git epydoc/docparser.py epydoc/docparser.py
index 929ad52..0d31fa1 100644
--- epydoc/docparser.py
+++ epydoc/docparser.py
@@ -564,17 +564,6 @@ def process_file(module_doc):
                 # indented line, with no final newline.
                 # (otherwise, this is the wrong thing to do.)
                 pass
-            
-        # Line-internal newline token: if we're still at the start of
-        # the logical line, and we've seen one or more comment lines,
-        # then discard them: blank lines are not allowed between a
-        # comment block and the thing it describes.
-        elif toktype == tokenize.NL:
-            if comments and not line_toks:
-                log.warning('Ignoring docstring comment block followed by '
-                            'a blank line in %r on line %r' %
-                            (module_doc.filename, srow-1))
-                comments = []
                 
         # Comment token: add to comments if appropriate.
         elif toktype == tokenize.COMMENT:
